### ISSUE TITLE
Public Read write access to the buckets where resources are uploaded. 

### ISSUE THREAD

**Issue by pankajvkulkarni** [aws-quickstart/taskcat#52]
Hi, 

I see that the buckets which are getting created as got public-read 

1. https://github.com/aws-quickstart/taskcat/blob/master/taskcat/stacker.py#L317
2. https://github.com/aws-quickstart/taskcat/blob/master/taskcat/stacker.py#L325

Also the objects which are being uploaded to the bucket has got public-read permission 

https://github.com/aws-quickstart/taskcat/blob/master/taskcat/stacker.py#L349

Is it necessary to have the "public-read" for the buckets and the objects ?

**Comment by avattathil** [comment]
It depends on how the cfn is structured. If your userdata is using curl to pull file then yes public read is needed.
You can also add a existing private bucket and IAM roles to your cfn. 
If the `s3bucket` is specified in the _config.yml_ then taskcat will not create a bucket. 
```
global:
  owner: owner@company.com
  project: sample-cloudformation-project <- Must match the root directory of project (usually the name of git repo)
  s3bucket: projectx-templates
```

**Comment by pankajvkulkarni** [comment]
I am using existing bucket,  I am using IAM profile to run the task cat the this role does not have "s3:PutObjectAcl". 

How about this for the existing bucket.
https://github.com/aws-quickstart/taskcat/blob/master/taskcat/stacker.py#L349

`s3_client.upload_file(filename, self.get_s3bucket(), upload, ExtraArgs={'ACL': 'public-read'})`

My account admin does not want me to grant public-read permissions on any objects in s3

**Comment by avattathil** [comment]
You can skip the upload by commenting the stage call as a workaround. 
https://github.com/aws-quickstart/taskcat/blob/master/taskcat.py#L43

**Comment by pankajvkulkarni** [comment]
ok, Thanks, 
Can you please make this as command line parameter where we skip upload  s3 and just test ?

**Comment by avattathil** [comment]
Yes good idea, I will add that to the feature roadmap

**Comment by avattathil** [comment]
Depends on https://github.com/aws-quickstart/taskcat/issues/59

**Comment by agamzn** [comment]
@pankajvkulkarni, 

The permissions issues have been addressed in pull #67, via issue #59. Your feature request for the additional parameter to skip S3 has been moved to #68 .

Closing this out.

**Comment by avattathil** [comment]
Issue Fixed in https://github.com/aws-quickstart/taskcat/pull/72 👍
